modules/genai-ecosystem/pages/langchain4j.adoc

```diff
@@ -19,27 +19,443 @@ The Neo4j Integration makes the xref:vector-search.adoc[Neo4j Vector] index avai
+        .apiKey(OPENAI_API_KEY)
+        .modelName(GPT_4_O_MINI)
+        .build();
+
+// Neo4jText2CypherRetriever instance
+Neo4jText2CypherRetriever retriever = Neo4jText2CypherRetriever.builder()
+        .graph(neo4jGraph)
+        .chatLanguageModel(chatLanguageModel)
+        .build();
+----
+
+== Usage Examples
+
+=== Neo4jEmbeddingStore
+
+We can define an `Neo4jEmbeddingStore` with the required Java Driver instance and dimension:
+
+[source,java]
+----
+Driver driver = GraphDatabase.driver('<Bolt URL>', AuthTokens.basic('<username>', '<password>'));
+
+Neo4jEmbeddingStore embeddingStore = Neo4jEmbeddingStore.builder()
+    .withDriver(driver)
+    .dimension(384)
+    .label("CustomLabel")
+    .build();
+----
+
+Or use withBasicAuth as alternative to `driver(<driver>)`
+[source,java]
+----
+
+// we can use with `withBasicAuth` as alternative to driver:
+Neo4jEmbeddingStore embeddingStore = Neo4jEmbeddingStore.builder().withBasicAuth("<Bolt URL>", "<username>", "<password>").build();
+----
+
+We can define an `Neo4jEmbeddingStore` with the required configurations and an optional label name in this way:
+[source,java]
+----
+Driver driver = GraphDatabase.driver('<Bolt URL>', AuthTokens.basic('<username>', '<password>'));
+
+Neo4jEmbeddingStore embeddingStoreWithCustomLabel = Neo4jEmbeddingStore.builder()
+    .withDriver(driver)
+    .dimension(384)
+    .label("CustomLabel")
+    .build();
+----
+
+
+To add a single embedding with a `TextSegment`:
+
+[source,java]
+----
+TextSegment segment = TextSegment.from("I like football.");
+Embedding embedding = embeddingModel.embed(segment.text()).content();
+String id = embeddingStore.add(embedding, segment);
+// output: id of the embedding
+----
+
+The above embedding creates a node with a label `Document` and the properties `embedding` (with the embedded text), `text` with the original text, `id` with a UUID: 
+
+image::defaultEmbeddingNode.png[]
+
+Add multiple embeddings:
+
+[source,java]
+----
+TextSegment segment = TextSegment.from(randomUUID());
+Embedding firstEmbedding = embeddingModel.embed("firstEmbedText").content();
+Embedding secondEmbedding = embeddingModel.embed("secondEmbedText").content();
+List<String> ids = embeddingStore.addAll(asList(firstEmbedding, secondEmbedding));
+// output: list of the embedding ids
+----
+
+
+To add embeddings with `TextSegment` and a metadata with key "foo" and value "bar":
+
+[source,java]
+----
+TextSegment segment = TextSegment.from(randomUUID(), Metadata.from("foo", "bar"));
+Embedding embedding = embeddingModel.embed(segment.text()).content();
+String id = embeddingStore.add(embedding, segment);
+// output: id of the embedding
+----
 
+To add embeddings with `TextSegment` and 3 metadata with keys "foo1", "foo2", "foo3" and values respectively "bar1", "bar2", "bar3":
 [source,java]
 ----
-include::https://github.com/langchain4j/langchain4j-examples/raw/main/neo4j-example/src/main/java/Neo4jEmbeddingStoreExample.java[]
```

Comment:
can we include tagged regions so we don't need to maintain two copies of the code?

Reply:
Ok, i'll add the `tag::`s in the `langchain4j-examples` files